### ISSUE TITLE
Javascript code in onerror attributes run forever

### ISSUE THREAD

**Issue by samuellb** [RequestPolicy/requestpolicy#441]
(Don't) Try this code in RequestPolicy:

```
<img src="http://test.invalid/" alt="" onerror="this.src='http://test2.invalid/'">
```

It will keep running the script in the onerror attribute forever, even after you close the tab (or even the browser). The only way to stop the code from running is to terminate the browser process.

According to the builtin profiler in Firefox it runs in the context of the web site fortunately (even after closing the tab).

This causes 100% CPU utilization on websites that use the MediaCreeper tracking code (mostly some swedish blogs).

I've tested this on RequestPolicy 0.5.28 on Iceweasel(=Firefox for Debian/Linux) 29.0.1

**Comment by samuellb** [comment]
Tested on 1.x and there it seems to have been fixed :)

**Comment by samuellb** [comment]
Actually it DOES happen on 1.x also... It turns out I had javascript disabled when I tested last time.

**Comment by myrdd** [comment]
@samuellb, please post some links so that I can test this.

**Comment by samuellb** [comment]
@myrdd, these web site have the "onerror" code and cause problems with RequestPolicy:
http://cornucopia.cornubot.se/
http://newsvoice.se/

**Comment by myrdd** [comment]
Okay, now I understand. By "invalid" you mean _blocked_. I assume that if the image gets blocked, firefox executes `onerror`.
There has to be some kind of loop detection. My first idea is: if the same node (e.g. `<img>`) gets blocked several times, then we might remove that node or the `onerror` argument.

Maybe the arguments of `shouldLoad` contain some useful information, I'll have a look at them. But maybe this is rather a firefox bug, as firefox requests the same URL several times. We'll see.

By the way, this is the code that causes problems. It's the same on both sites.

``` html
<img
    src="http://mediacreeper.com/image"
    alt="MediaCreeper"
    onerror="this.src='http://mediacreeper.se/image';"
    style="width: 80px;height: 15px;border: 0px;"
/>
```

**Comment by SkySkimmer** [comment]
Sounds like this bug https://bugzilla.mozilla.org/show_bug.cgi?id=528448 to me.

**Comment by myrdd** [comment]
Yes, imho this is a firefox bug, so this is could be tagged `wontfix` – sorry.
@SkySkimmer Thanks for the link, I've posted there a reference to this bug.

**Comment by myrdd** [comment]
@samuellb have a look at https://github.com/RequestPolicyContinued/requestpolicy/issues/441#issuecomment-60025118
